code/game/machinery/computer/HolodeckControl.dm

```diff
@@ -230,6 +230,9 @@
 	for(var/obj/effect/decal/cleanable/blood/B in linkedholodeck)
 		qdel(B)
 
+	for(var/obj/effect/decal/cleanable/blooddrip/D in linkedholodeck)

```

Comment:
Both checks here and the carp check under it should get torn out, they're frankly ridiculous.